### ISSUE TITLE
Improve F1 help for tokens with multiple uses

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#48392]
There are a number of keywords and tokens in C# that have different meanings based on context and in a lot of cases our F1 documentation doesn't draw any distinction between different usages, despite there being docs pages written for each use.

This work would be done in conjunction with, or after https://github.com/dotnet/docs/issues/20799 has been completed, to ensure unique `f1_keyword` metadata is present in the docs pages, and then update Roslyn to send the correct `f1_keyword` in the right context.

Rather than log issues about each individual token, https://github.com/dotnet/docs/issues/20799 should be used as a checklist of items to work through. Anyone contributing fixes in this area should do so only after the docs have been updated accordingly, or should open a PR in the docs repo at the same time.

For example, the `!` token was ambiguous in the docs so first https://github.com/dotnet/docs/pull/20341 was created to fix the documentation and then https://github.com/dotnet/roslyn/pull/47516 was created to fix Roslyn. These two PRs form a very good basis for future work.

**Comment by Youssef1313** [comment]
@davidwengier Should we have another issue to track F1 service for codestyle options?
See https://github.com/dotnet/docs/pull/20892#discussion_r499620992

I tried to give it a look. Here is the IHelpContextService interface.

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/VisualStudio/Core/Def/Implementation/F1Help/IHelpContextService.cs

Its method receives a `Microsoft.CodeAnalysis.Document`, which AFAIK is for C# and VB only. However, the .editorconfig file is in `ini` format.

**Comment by davidwengier** [comment]
> Should we have another issue

Yes, that seems unrelated to this one.

**Comment by CyrusNajmabadi** [comment]
@davidwengier anything left to do here?

**Comment by davidwengier** [comment]
Both docs issues linked above still seem to be open, and lots of unchecked checkmarks, but if its not like anything is not working, so feel free to close if you want to clean up open issues.

**Comment by CyrusNajmabadi** [comment]
We woudl take community PRs here.

**Comment by CyrusNajmabadi** [comment]
Closing out as speculative.  IF we need this, we should assign to dev/milestone to have it be done.  PRs are fine.  But this tracking issue isn't moving hte needle on that.
